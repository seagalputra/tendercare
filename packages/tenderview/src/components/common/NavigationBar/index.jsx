import React from 'react'
import { useHistory } from 'react-router-dom'
import { makeStyles } from '@material-ui/core/styles'
import {
  AppBar,
  Toolbar,
  Typography,
  Button,
  IconButton
} from '@material-ui/core'
import MenuIcon from '@material-ui/icons/Menu'

import { isLoggedIn, clearToken } from 'utils/sessions'

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1
  },
  navigation: {
    boxShadow: 'none'
  },
  menuButton: {
    marginRight: theme.spacing(2)
  },
  title: {
    flexGrow: 1
  }
}))

const NavigationBar = () => {
  const classes = useStyles()
  const history = useHistory()

  const handleLogout = () => {
    clearToken()
    history.push('/')
  }

  return (
    <header className={classes.root}>
      <AppBar
        className={classes.navigation}
        position="static"
        color="transparent"
      >
        <Toolbar>
          <IconButton
            edge="start"
            className={classes.menuButton}
            color="inherit"
            aria-label="menu"
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" className={classes.title} color="inherit">
            Tendercare
          </Typography>
          {isLoggedIn() ? (
            <Button color="inherit" onClick={handleLogout}>
              Keluar
            </Button>
          ) : null}
        </Toolbar>
      </AppBar>
    </header>
  )
}

export default NavigationBar
