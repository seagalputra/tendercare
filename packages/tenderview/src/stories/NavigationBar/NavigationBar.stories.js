import React from 'react'
import NavigationBar from 'components/common/NavigationBar'

export default {
  title: 'Navigation Bar',
  component: NavigationBar,
  excudeStories: /.*Data$/
}

export const Default = () => <NavigationBar />
