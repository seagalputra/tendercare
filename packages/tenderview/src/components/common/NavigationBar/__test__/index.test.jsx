import React from 'react'
import { shallow } from 'enzyme'
import NavigationBar from '..'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: jest.fn()
  })
}))

describe('NavigationBar', () => {
  it('renders without crashing', () => {
    const component = shallow(<NavigationBar />)
    expect(component).toMatchSnapshot()
  })
})
