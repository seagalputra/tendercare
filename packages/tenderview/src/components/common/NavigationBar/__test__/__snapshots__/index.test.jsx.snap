// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders without crashing 1`] = `
<header
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-navigation-2"
    color="transparent"
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="menu"
        className="makeStyles-menuButton-3"
        color="inherit"
        edge="start"
      >
        <MenuIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-4"
        color="inherit"
        variant="h6"
      >
        Tendercare
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</header>
`;
